datafusion-cli/src/main.rs

```diff
@@ -109,7 +125,32 @@ pub async fn main() -> Result<()> {
         session_config = session_config.with_batch_size(batch_size);
     };
 
-    let runtime_env = create_runtime_env()?;
+    let rn_config = RuntimeConfig::new();
+    let rn_config =
+        // set memory pool size
+        if let Some(memory_limit) = args.memory_limit {
+            let memory_limit = memory_limit[..memory_limit.len() - 1]
```

Comment:
I think this simply passes `10` (bytes) along to the pool size when `-m 10G` is passed:

```shell
$ RUST_LOG=debug /Users/alamb/Software/target-df/debug/datafusion-cli -m 10g
DataFusion CLI v30.0.0
[2023-08-26T11:06:26Z DEBUG datafusion_execution::memory_pool::pool] Created new GreedyMemoryPool(pool_size=10)
```


I think  the pool size should work like this. What do you think?

* `-m 1000` -- set pool size to 1000
* `-m 500m` -- set pool size to `500 * 1024*1024` (500 MB)
* `-m 10G` -- set pool size to `10 * 1024*1024*1024` (10 GB)